lib/tbot/config/config.go

```diff
@@ -159,6 +164,37 @@ type OnboardingConfig struct {
 	JoinMethod types.JoinMethod `yaml:"join_method"`
 }
 
+// HasTokenValue gives the ability to check if there has been a token value stored
+// in the config
+func (conf *OnboardingConfig) HasTokenValue() bool {
+	return conf.TokenValue != ""
+}
+
+// StoreToken stores the value for --token or auth_token in the config
+//
+// In the case of the token value pointing to a file, this allows us to
+// fetch the value of the token when it's needed (when connecting for the first time)
+// instead of trying to read the file every time that teleport is launched.
+// This means we can allow temporary token files that are removed after teleport has
+// successfully connected the first time.
+func (conf *OnboardingConfig) StoreToken(token string) {
+	conf.TokenValue = token
+}
+
+// GetToken returns token needed to join the auth server
+//
+// If the value stored points to a file, it will attempt to read the token value from the file
+// and return an error if it wasn't successful
+// If the value stored doesn't point to a file, it'll return the value stored
+func (conf *OnboardingConfig) GetToken() (string, error) {
+	token, err := utils.TryReadValueAsFile(conf.TokenValue)
```

Comment:
Can we remove this method? It's pretty much identical to `TryReadValueAsFile`, so we could just use that instead.

Reply:
Personality I'd rather keep it as it provides a nice way to grab the token without caring about how it's retrieved.